System runlevel triggered rules trigger before done being saved

<!-- Provide a general summary of the issue in the *Title* above -->

<!-- Important: Please contact the openHAB community forum for questions or -->
<!-- for configuration and usage guidance: https://community.openhab.org -->

<!-- Feel free to delete any comment lines in the template (starting with "<!--") -->

## Expected Behavior
<!-- If you're describing a bug, tell us what should happen -->
<!-- If you're suggesting a change/improvement, tell us how it should work -->

Given the recent changes to system started/runlevel triggers so they *always* trigger if that runlevel has passed I would expect a UI rule to run after the "save" button is pressed.

## Current Behavior
<!-- If describing a bug, tell us what happens instead of the expected behavior -->
<!-- Include related log information (preferably debug level) and related configs -->
<!-- Use a file attachment for log and config information longer than a few lines -->
<!-- Enclose multi-line log/code snippets with ``` on new lines for proper formatting -->
<!-- If suggesting a change/improvement, explain the difference from current behavior -->
<!-- For improvements, discuss at community.openhab.org first and include link to topic -->

After pressing "save" the rule attempts to run but because it's not yet ready an error is produced in the logs. 

```
2023-07-19 09:00:16.577 [ERROR] [e.automation.internal.RuleEngineImpl] - Failed to execute rule ‘time-statemachine' with status 'INITIALIZING'
```

## Possible Solution
<!-- Not obligatory, but suggest a fix/reason for the bug, -->
<!-- or ideas how to implement the addition or change -->

Wait for the rule to become IDLE before attempting to trigger it or go back to the old behavior and don't run the rule at all on a save.

## Steps to Reproduce (for Bugs)
<!-- Provide a link to a live example, or an unambiguous set of steps to -->
<!-- reproduce this bug. Include code to reproduce, if relevant -->
1. Create a rule in the UI (I'm seeing this with my Time Based State Machine rule template so you could use that if you want)
2. Open the Action and make a change.
3. Click "Save" and watch the error appear in the logs.

## Context
<!-- How has this issue affected you? What are you trying to accomplish? -->
<!-- Providing context helps us come up with a solution that is most useful in the real world -->

Errors in the logs while doing "normal" actions are alarming and will generate lots of forum traffic. 

Without watching the logs it is not apparent that the rule failed to run as expected. 

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->

* Version used: (e.g., openHAB and add-on versions) Snapshot #3538
* Environment name and version (e.g. Chrome 111, Java 17, Node.js 18.15, ...): Docker version
* Operating System and version (desktop or mobile, Windows 11, Raspbian Bullseye, ...): Docker version
```
runtimeInfo:
  version: 4.0.0
  buildString: "Build #3538"
locale: en-US
systemInfo:
  configFolder: /openhab/conf
  userdataFolder: /openhab/userdata
  logFolder: /openhab/userdata/logs
  javaVersion: 17.0.7
  javaVendor: Debian
  osName: Linux
  osVersion: 5.15.0-76-generic
  osArchitecture: amd64
  availableProcessors: 4
  freeMemory: 356040672
  totalMemory: 874512384
  startLevel: 100
bindings: null
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: false
    macos: true
    windows: false
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    os: macos
    pixelRatio: 2
    prefersColorScheme: light
  isSecureContext: false
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: N/A
    hardwareConcurrency: 12
    language: en-US
    languages:
      - en-US
      - en
    onLine: true
    platform: MacIntel
  screen:
    width: 2560
    height: 1440
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: light
    filled: true
    pageTransitionAnimation: default
    bars: filled
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36
    (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
timestamp: 2023-07-19T15:07:38.366Z
```

--- a/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleEngineImpl.java
+++ b/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleEngineImpl.java
@@ -155,6 +155,7 @@ public class RuleEngineImpl implements RuleManager, RegistryChangeListener<Modul
      * The storage for the disable information
      */
     private final Storage<Boolean> disabledRulesStorage;
+    private final StartLevelService startLevelService;
 
     /**
      * Locker which does not permit rule initialization when the rule engine is stopping.
@@ -254,7 +255,7 @@ public class RuleEngineImpl implements RuleManager, RegistryChangeListener<Modul
     @Activate
     public RuleEngineImpl(final @Reference ModuleTypeRegistry moduleTypeRegistry,
             final @Reference RuleRegistry ruleRegistry, final @Reference StorageService storageService,
-            final @Reference ReadyService readyService) {
+            final @Reference ReadyService readyService, final @Reference StartLevelService startLevelService) {
         this.disabledRulesStorage = storageService.<Boolean> getStorage(DISABLED_RULE_STORAGE,
                 this.getClass().getClassLoader());
 
@@ -265,6 +266,7 @@ public class RuleEngineImpl implements RuleManager, RegistryChangeListener<Modul
 
         this.ruleRegistry = ruleRegistry;
         this.readyService = readyService;
+        this.startLevelService = startLevelService;
 
         listener = new RegistryChangeListener<>() {
             @Override
@@ -846,6 +848,17 @@ public class RuleEngineImpl implements RuleManager, RegistryChangeListener<Modul
         // Register the rule and set idle status.
         register(rule);
         setStatus(ruleUID, new RuleStatusInfo(RuleStatus.IDLE));
+
+        // check if we have to trigger because of the startlevel
+        List<Trigger> slTriggers = rule.getTriggers().stream().map(WrappedTrigger::unwrap)
+                .filter(t -> SystemTriggerHandler.STARTLEVEL_MODULE_TYPE_ID.equals(t.getTypeUID())).toList();
+        if (slTriggers.stream()
+                .anyMatch(t -> ((BigDecimal) t.getConfiguration().get(SystemTriggerHandler.CFG_STARTLEVEL))
+                        .intValue() <= startLevelService.getStartLevel())) {
+            runNow(rule.getUID(), true,
+                    Map.of(SystemTriggerHandler.OUT_STARTLEVEL, StartLevelService.STARTLEVEL_RULES));
+        }
+
         return true;
     }
 

--- a/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/factory/CoreModuleHandlerFactory.java
+++ b/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/factory/CoreModuleHandlerFactory.java
@@ -116,7 +116,7 @@ public class CoreModuleHandlerFactory extends BaseModuleHandlerFactory implement
             } else if (ItemCommandTriggerHandler.MODULE_TYPE_ID.equals(moduleTypeUID)) {
                 return new ItemCommandTriggerHandler(trigger, ruleUID, bundleContext, itemRegistry);
             } else if (SystemTriggerHandler.STARTLEVEL_MODULE_TYPE_ID.equals(moduleTypeUID)) {
-                return new SystemTriggerHandler(trigger, bundleContext, startLevelService);
+                return new SystemTriggerHandler(trigger, bundleContext);
             } else if (ThingStatusTriggerHandler.CHANGE_MODULE_TYPE_ID.equals(moduleTypeUID)
                     || ThingStatusTriggerHandler.UPDATE_MODULE_TYPE_ID.equals(moduleTypeUID)) {
                 return new ThingStatusTriggerHandler(trigger, bundleContext);

--- a/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/SystemTriggerHandler.java
+++ b/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/SystemTriggerHandler.java
@@ -47,15 +47,13 @@ public class SystemTriggerHandler extends BaseTriggerModuleHandler implements Ev
 
     private final Integer startlevel;
     private final Set<String> types;
-    private final StartLevelService startLevelService;
 
     private boolean triggered = false;
 
     private final ServiceRegistration<?> eventSubscriberRegistration;
 
-    public SystemTriggerHandler(Trigger module, BundleContext bundleContext, StartLevelService startLevelService) {
+    public SystemTriggerHandler(Trigger module, BundleContext bundleContext) {
         super(module);
-        this.startLevelService = startLevelService;
         this.startlevel = ((BigDecimal) module.getConfiguration().get(CFG_STARTLEVEL)).intValue();
         if (STARTLEVEL_MODULE_TYPE_ID.equals(module.getTypeUID())) {
             this.types = Set.of(StartlevelEvent.TYPE);
@@ -69,12 +67,6 @@ public class SystemTriggerHandler extends BaseTriggerModuleHandler implements Ev
     @Override
     public void setCallback(ModuleHandlerCallback callback) {
         super.setCallback(callback);
-
-        // trigger immediately when start level is already reached
-        int currentStartLevel = startLevelService.getStartLevel();
-        if (currentStartLevel > StartLevelService.STARTLEVEL_RULEENGINE && currentStartLevel >= startlevel) {
-            trigger();
-        }
     }
 
     @Override

--- a/bundles/org.openhab.core.automation/src/test/java/org/openhab/core/automation/internal/module/handler/SystemTriggerHandlerTest.java
+++ b/bundles/org.openhab.core.automation/src/test/java/org/openhab/core/automation/internal/module/handler/SystemTriggerHandlerTest.java
@@ -67,34 +67,17 @@ public class SystemTriggerHandlerTest {
     public void testDoesNotTriggerIfStartLevelTooLow() {
         when(startLevelServiceMock.getStartLevel()).thenReturn(0);
 
-        SystemTriggerHandler triggerHandler = new SystemTriggerHandler(triggerMock, bundleContextMock,
-                startLevelServiceMock);
+        SystemTriggerHandler triggerHandler = new SystemTriggerHandler(triggerMock, bundleContextMock);
         triggerHandler.setCallback(callbackMock);
 
         verifyNoInteractions(callbackMock);
     }
 
     @Test
-    public void testDoesTriggerImmediatelyIfStartLevelHigherOnInit() {
-        when(startLevelServiceMock.getStartLevel()).thenReturn(100);
-
-        SystemTriggerHandler triggerHandler = new SystemTriggerHandler(triggerMock, bundleContextMock,
-                startLevelServiceMock);
-        triggerHandler.setCallback(callbackMock);
-
-        ArgumentCaptor<Map> captor = ArgumentCaptor.forClass(Map.class);
-        verify(callbackMock).triggered(eq(triggerMock), captor.capture());
-
-        Map<String, Object> configuration = (Map<String, Object>) captor.getValue();
-        assertThat(configuration.get(SystemTriggerHandler.OUT_STARTLEVEL), is(CFG_STARTLEVEL));
-    }
-
-    @Test
     public void testDoesNotTriggerIfStartLevelEventLower() {
         when(startLevelServiceMock.getStartLevel()).thenReturn(0);
 
-        SystemTriggerHandler triggerHandler = new SystemTriggerHandler(triggerMock, bundleContextMock,
-                startLevelServiceMock);
+        SystemTriggerHandler triggerHandler = new SystemTriggerHandler(triggerMock, bundleContextMock);
         triggerHandler.setCallback(callbackMock);
 
         Event event = SystemEventFactory.createStartlevelEvent(70);
@@ -107,8 +90,7 @@ public class SystemTriggerHandlerTest {
     public void testDoesTriggerIfStartLevelEventHigher() {
         when(startLevelServiceMock.getStartLevel()).thenReturn(0);
 
-        SystemTriggerHandler triggerHandler = new SystemTriggerHandler(triggerMock, bundleContextMock,
-                startLevelServiceMock);
+        SystemTriggerHandler triggerHandler = new SystemTriggerHandler(triggerMock, bundleContextMock);
         triggerHandler.setCallback(callbackMock);
 
         Event event = SystemEventFactory.createStartlevelEvent(100);
@@ -119,34 +101,13 @@ public class SystemTriggerHandlerTest {
 
         Map<String, Object> configuration = (Map<String, Object>) captor.getValue();
         assertThat(configuration.get(SystemTriggerHandler.OUT_STARTLEVEL), is(CFG_STARTLEVEL));
-    }
-
-    @Test
-    public void testDoesNotTriggerAfterInitialTrigger() {
-        when(startLevelServiceMock.getStartLevel()).thenReturn(100);
-
-        SystemTriggerHandler triggerHandler = new SystemTriggerHandler(triggerMock, bundleContextMock,
-                startLevelServiceMock);
-        triggerHandler.setCallback(callbackMock);
-
-        ArgumentCaptor<Map> captor = ArgumentCaptor.forClass(Map.class);
-        verify(callbackMock).triggered(eq(triggerMock), captor.capture());
-
-        Map<String, Object> configuration = (Map<String, Object>) captor.getValue();
-        assertThat(configuration.get(SystemTriggerHandler.OUT_STARTLEVEL), is(CFG_STARTLEVEL));
-
-        Event event = SystemEventFactory.createStartlevelEvent(100);
-        triggerHandler.receive(event);
-
-        verifyNoMoreInteractions(callbackMock);
     }
 
     @Test
     public void testDoesNotTriggerAfterEventTrigger() {
         when(startLevelServiceMock.getStartLevel()).thenReturn(0);
 
-        SystemTriggerHandler triggerHandler = new SystemTriggerHandler(triggerMock, bundleContextMock,
-                startLevelServiceMock);
+        SystemTriggerHandler triggerHandler = new SystemTriggerHandler(triggerMock, bundleContextMock);
         triggerHandler.setCallback(callbackMock);
 
         Event event = SystemEventFactory.createStartlevelEvent(100);

--- a/itests/org.openhab.core.automation.integration.tests/src/main/java/org/openhab/core/automation/integration/test/AutomationIntegrationJsonTest.java
+++ b/itests/org.openhab.core.automation.integration.tests/src/main/java/org/openhab/core/automation/integration/test/AutomationIntegrationJsonTest.java
@@ -16,6 +16,7 @@ import static org.hamcrest.CoreMatchers.*;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 import java.util.Collection;
 import java.util.Optional;
@@ -80,6 +81,7 @@ public class AutomationIntegrationJsonTest extends JavaOSGiTest {
     private final Logger logger = LoggerFactory.getLogger(AutomationIntegrationJsonTest.class);
     private @NonNullByDefault({}) EventPublisher eventPublisher;
     private @NonNullByDefault({}) ItemRegistry itemRegistry;
+    private @NonNullByDefault({}) StartLevelService startLevelService;
     private @NonNullByDefault({}) RuleRegistry ruleRegistry;
     private @NonNullByDefault({}) RuleManager ruleManager;
     private @NonNullByDefault({}) ManagedRuleProvider managedRuleProvider;
@@ -96,8 +98,12 @@ public class AutomationIntegrationJsonTest extends JavaOSGiTest {
 
         eventPublisher = getService(EventPublisher.class);
         itemRegistry = getService(ItemRegistry.class);
+        startLevelService = mock(StartLevelService.class);
+        when(startLevelService.getStartLevel()).thenReturn(100);
+        registerService(startLevelService, StartLevelService.class.getName());
+
         CoreModuleHandlerFactory coreModuleHandlerFactory = new CoreModuleHandlerFactory(getBundleContext(),
-                eventPublisher, itemRegistry, mock(TimeZoneProvider.class), mock(StartLevelService.class));
+                eventPublisher, itemRegistry, mock(TimeZoneProvider.class), startLevelService);
         mock(CoreModuleHandlerFactory.class);
         registerService(coreModuleHandlerFactory);
 

--- a/itests/org.openhab.core.automation.integration.tests/src/main/java/org/openhab/core/automation/integration/test/AutomationIntegrationTest.java
+++ b/itests/org.openhab.core.automation.integration.tests/src/main/java/org/openhab/core/automation/integration/test/AutomationIntegrationTest.java
@@ -17,6 +17,7 @@ import static org.hamcrest.CoreMatchers.*;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 import java.util.Collection;
 import java.util.Collections;
@@ -98,6 +99,7 @@ public class AutomationIntegrationTest extends JavaOSGiTest {
     private final Logger logger = LoggerFactory.getLogger(AutomationIntegrationTest.class);
     private @Nullable EventPublisher eventPublisher;
     private @Nullable ItemRegistry itemRegistry;
+    private @NonNullByDefault({}) StartLevelService startLevelService;
     private @Nullable RuleRegistry ruleRegistry;
     private @Nullable RuleManager ruleEngine;
     private @Nullable ManagedRuleProvider managedRuleProvider;
@@ -113,9 +115,13 @@ public class AutomationIntegrationTest extends JavaOSGiTest {
 
         eventPublisher = getService(EventPublisher.class);
         itemRegistry = getService(ItemRegistry.class);
+        startLevelService = mock(StartLevelService.class);
+        when(startLevelService.getStartLevel()).thenReturn(100);
+        registerService(startLevelService, StartLevelService.class.getName());
+
         CoreModuleHandlerFactory coreModuleHandlerFactory = new CoreModuleHandlerFactory(getBundleContext(),
                 Objects.requireNonNull(eventPublisher), Objects.requireNonNull(itemRegistry),
-                mock(TimeZoneProvider.class), mock(StartLevelService.class));
+                mock(TimeZoneProvider.class), startLevelService);
         mock(CoreModuleHandlerFactory.class);
         registerService(coreModuleHandlerFactory);
 

--- a/itests/org.openhab.core.automation.integration.tests/src/main/java/org/openhab/core/automation/integration/test/RuleSimulationTest.java
+++ b/itests/org.openhab.core.automation.integration.tests/src/main/java/org/openhab/core/automation/integration/test/RuleSimulationTest.java
@@ -15,6 +15,8 @@ package org.openhab.core.automation.integration.test;
 import static org.hamcrest.CoreMatchers.*;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
@@ -54,6 +56,7 @@ import org.openhab.core.automation.util.RuleBuilder;
 import org.openhab.core.common.registry.ProviderChangeListener;
 import org.openhab.core.config.core.Configuration;
 import org.openhab.core.service.ReadyMarker;
+import org.openhab.core.service.StartLevelService;
 import org.openhab.core.storage.StorageService;
 import org.openhab.core.test.java.JavaOSGiTest;
 import org.slf4j.Logger;
@@ -71,10 +74,15 @@ public class RuleSimulationTest extends JavaOSGiTest {
 
     private @Nullable RuleRegistry ruleRegistry;
     private @Nullable RuleManager ruleEngine;
+    private @NonNullByDefault({}) StartLevelService startLevelService;
 
     @BeforeEach
     public void before() {
         registerVolatileStorageService();
+
+        startLevelService = mock(StartLevelService.class);
+        when(startLevelService.getStartLevel()).thenReturn(100);
+        registerService(startLevelService, StartLevelService.class.getName());
 
         StorageService storageService = getService(StorageService.class);
 

--- a/itests/org.openhab.core.automation.module.core.tests/src/main/java/org/openhab/core/automation/internal/module/RunRuleModuleTest.java
+++ b/itests/org.openhab.core.automation.module.core.tests/src/main/java/org/openhab/core/automation/internal/module/RunRuleModuleTest.java
@@ -15,6 +15,7 @@ package org.openhab.core.automation.internal.module;
 import static java.util.Map.entry;
 import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 import java.util.ArrayList;
 import java.util.Collection;
@@ -69,9 +70,13 @@ public class RunRuleModuleTest extends JavaOSGiTest {
 
     private final Logger logger = LoggerFactory.getLogger(RunRuleModuleTest.class);
     private final VolatileStorageService volatileStorageService = new VolatileStorageService();
+    private @NonNullByDefault({}) StartLevelService startLevelService;
 
     @BeforeEach
     public void before() {
+        startLevelService = mock(StartLevelService.class);
+        when(startLevelService.getStartLevel()).thenReturn(100);
+        registerService(startLevelService, StartLevelService.class.getName());
         EventPublisher eventPublisher = getService(EventPublisher.class);
         ItemRegistry itemRegistry = getService(ItemRegistry.class);
         CoreModuleHandlerFactory coreModuleHandlerFactory = new CoreModuleHandlerFactory(getBundleContext(),

--- a/itests/org.openhab.core.automation.module.core.tests/src/main/java/org/openhab/core/automation/internal/module/RuntimeRuleTest.java
+++ b/itests/org.openhab.core.automation.module.core.tests/src/main/java/org/openhab/core/automation/internal/module/RuntimeRuleTest.java
@@ -15,6 +15,7 @@ package org.openhab.core.automation.internal.module;
 import static java.util.Map.entry;
 import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 import java.util.Collection;
 import java.util.Collections;
@@ -79,9 +80,13 @@ public class RuntimeRuleTest extends JavaOSGiTest {
 
     private final Logger logger = LoggerFactory.getLogger(RuntimeRuleTest.class);
     private final VolatileStorageService volatileStorageService = new VolatileStorageService();
+    private @NonNullByDefault({}) StartLevelService startLevelService;
 
     @BeforeEach
     public void before() {
+        startLevelService = mock(StartLevelService.class);
+        when(startLevelService.getStartLevel()).thenReturn(100);
+        registerService(startLevelService, StartLevelService.class.getName());
         EventPublisher eventPublisher = getService(EventPublisher.class);
         ItemRegistry itemRegistry = getService(ItemRegistry.class);
         CoreModuleHandlerFactory coreModuleHandlerFactory = new CoreModuleHandlerFactory(getBundleContext(),

--- a/itests/org.openhab.core.automation.module.timer.tests/src/main/java/org/openhab/core/automation/module/timer/internal/BasicConditionHandlerTest.java
+++ b/itests/org.openhab.core.automation.module.timer.tests/src/main/java/org/openhab/core/automation/module/timer/internal/BasicConditionHandlerTest.java
@@ -16,6 +16,7 @@ import static org.hamcrest.CoreMatchers.*;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 import java.util.ArrayList;
 import java.util.Collection;
@@ -78,6 +79,7 @@ public abstract class BasicConditionHandlerTest extends JavaOSGiTest {
     private @NonNullByDefault({}) RuleRegistry ruleRegistry;
     private @NonNullByDefault({}) RuleManager ruleEngine;
     private @Nullable Event itemEvent;
+    private @NonNullByDefault({}) StartLevelService startLevelService;
 
     /**
      * This executes before every test and before the
@@ -86,6 +88,9 @@ public abstract class BasicConditionHandlerTest extends JavaOSGiTest {
      */
     @BeforeEach
     public void beforeBase() {
+        startLevelService = mock(StartLevelService.class);
+        when(startLevelService.getStartLevel()).thenReturn(100);
+        registerService(startLevelService, StartLevelService.class.getName());
         EventPublisher eventPublisher = getService(EventPublisher.class);
         ItemRegistry itemRegistry = getService(ItemRegistry.class);
         CoreModuleHandlerFactory coreModuleHandlerFactory = new CoreModuleHandlerFactory(getBundleContext(),

--- a/itests/org.openhab.core.automation.module.timer.tests/src/main/java/org/openhab/core/automation/module/timer/internal/RuntimeRuleTest.java
+++ b/itests/org.openhab.core.automation.module.timer.tests/src/main/java/org/openhab/core/automation/module/timer/internal/RuntimeRuleTest.java
@@ -16,6 +16,7 @@ import static org.hamcrest.CoreMatchers.*;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 import java.util.Collection;
 import java.util.HashMap;
@@ -75,9 +76,13 @@ public class RuntimeRuleTest extends JavaOSGiTest {
     private VolatileStorageService volatileStorageService = new VolatileStorageService();
     private @NonNullByDefault({}) RuleRegistry ruleRegistry;
     private @NonNullByDefault({}) RuleManager ruleEngine;
+    private @NonNullByDefault({}) StartLevelService startLevelService;
 
     @BeforeEach
     public void before() {
+        startLevelService = mock(StartLevelService.class);
+        when(startLevelService.getStartLevel()).thenReturn(100);
+        registerService(startLevelService, StartLevelService.class.getName());
         EventPublisher eventPublisher = getService(EventPublisher.class);
         ItemRegistry itemRegistry = getService(ItemRegistry.class);
         CoreModuleHandlerFactory coreModuleHandlerFactory = new CoreModuleHandlerFactory(getBundleContext(),

--- a/itests/org.openhab.core.automation.tests/src/main/java/org/openhab/core/automation/event/RuleEventTest.java
+++ b/itests/org.openhab.core.automation.tests/src/main/java/org/openhab/core/automation/event/RuleEventTest.java
@@ -17,6 +17,7 @@ import static org.hamcrest.CoreMatchers.*;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 import java.util.ArrayList;
 import java.util.Collection;
@@ -78,12 +79,16 @@ public class RuleEventTest extends JavaOSGiTest {
 
     private @Nullable Event itemEvent = null;
     private @Nullable Event ruleRemovedEvent = null;
+    private @NonNullByDefault({}) StartLevelService startLevelService;
 
     public RuleEventTest() {
     }
 
     @BeforeEach
     public void before() {
+        startLevelService = mock(StartLevelService.class);
+        when(startLevelService.getStartLevel()).thenReturn(100);
+        registerService(startLevelService, StartLevelService.class.getName());
         EventPublisher eventPublisher = getService(EventPublisher.class);
         ItemRegistry itemRegistry = getService(ItemRegistry.class);
         CoreModuleHandlerFactory coreModuleHandlerFactory = new CoreModuleHandlerFactory(getBundleContext(),

--- a/itests/org.openhab.core.automation.tests/src/main/java/org/openhab/core/automation/internal/RuleEngineTest.java
+++ b/itests/org.openhab.core.automation.tests/src/main/java/org/openhab/core/automation/internal/RuleEngineTest.java
@@ -13,6 +13,8 @@
 package org.openhab.core.automation.internal;
 
 import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 import java.math.BigDecimal;
 import java.util.ArrayList;
@@ -41,6 +43,7 @@ import org.openhab.core.config.core.ConfigDescriptionParameterBuilder;
 import org.openhab.core.config.core.Configuration;
 import org.openhab.core.config.core.FilterCriteria;
 import org.openhab.core.config.core.ParameterOption;
+import org.openhab.core.service.StartLevelService;
 import org.openhab.core.test.java.JavaOSGiTest;
 
 /**
@@ -54,10 +57,14 @@ public class RuleEngineTest extends JavaOSGiTest {
 
     private @NonNullByDefault({}) RuleEngineImpl ruleEngine;
     private @NonNullByDefault({}) RuleRegistry ruleRegistry;
+    private @NonNullByDefault({}) StartLevelService startLevelService;
 
     @BeforeEach
     public void setup() {
         registerVolatileStorageService();
+        startLevelService = mock(StartLevelService.class);
+        when(startLevelService.getStartLevel()).thenReturn(100);
+        registerService(startLevelService, StartLevelService.class.getName());
         ruleEngine = (RuleEngineImpl) getService(RuleManager.class);
         ruleRegistry = getService(RuleRegistry.class);
         registerService(new TestModuleTypeProvider(), ModuleTypeProvider.class.getName());

